atp/client.go

```diff
@@ -25,6 +25,8 @@ type Client interface {
 	ReadSchema() (schema.Schema[schema.Step], error)
 	// Execute executes a step with a given context and returns the resulting output.
 	Execute(ctx context.Context, stepID string, input any) (outputID string, outputData any, err error)
+	Encoder() *cbor.Encoder
+	Decoder() *cbor.Decoder
```

Comment:
Is this necessary outside of testing? It's not always a good idea to expose everything.

Reply:
Afaik, it is not necessary outside of testing. It appeared to be the only solution to create deterministic edge cases.

What do you suggest instead?